lib/qif/writer.rb

```diff
@@ -61,11 +62,17 @@ def initialize(io, type = 'Bank', format = 'dd/mm/yyyy')
     
     # Add a transaction for writing
     def <<(transaction)
-      @transactions << transaction
+      case transaction.class.to_s
```

Comment:
This did cross my mind when I wrote this. According to the QIF spec:

>The account header !Account is used in two places-at the start of an account list and the start of a list of transactions to specify to which account they belong. 
via http://www.respmech.com/mym2qifw/qif_new.htm

My take on the first use "at the start of an account list" is for when the QIF file contains an account structure and no transaction data, therefore not relevant.

In the second use case, at the "start of a list of transactions to specify to which account they belong", the !Account block is used to describe the subsequent transactions. Therefore the subsequent Transactions belong to the Account. I have seen multiple Account blocks in a QIF file where the transactions appear sequentially for multiple accounts. Think of the Account section being like a header for the transaction records.

To implement the above would need quite a re-write to both Account and Transaction classes and associated reader and writer methods.

For what I am working on I only need one Account per file but will look into making the other changes.